lib/tasks/duplicate_labware_data_fix.rake

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+namespace :duplicates do
+  desc "De-duplicate labware"
+  task delete: :environment do |_t|
+    ActiveRecord::Base.transaction do
+      puts "Starting de-duping..."
+
+      # Get all the barcodes where there are duplicates
+      barcode_to_count = Labware.group(:barcode).count
+
+      barcode_to_count_duped = barcode_to_count.select { |_barcode, count| count > 1 } # 7106
+      puts "Barcodes with duplicates: #{barcode_to_count_duped.size}"
+
+      labs = Labware.where(barcode: barcode_to_count_duped.keys) # 19,240 (between 2 and 3 times above)
+      # of which, created today = 19,233 and before today = 7
+      puts "Labware with those barcodes: #{labs.size}"
+
+      barcode_to_labs = labs.group_by(&:barcode)
+
+      puts "Deleting duplicates..."
+      barcode_to_labs.each do |_barcode, labwares|
+        # save one of them
+        labwares.pop
+
+        # delete the rest
+        labwares.each do |lab|
+          lab.audits.each(&:destroy)
```

Comment:
you can do
`lab.audits.destroy_all`

Reply:
Cool thanks for comments. Won't change this time as it doesn't really matter for a rake task, but will bear in mind for future.